src/app/lib/business/forced_update/update_required_service.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:async';
+
+import 'package:app/access/forced_update/current_version_repository.dart';
+import 'package:app/access/forced_update/minimum_version_repository.dart';
+
+/// This service checks if the application is running the minimum required version.
+abstract interface class UpdateRequiredService {
+  factory UpdateRequiredService(
+    MinimumVersionRepository minimumVersionRepository,
+    CurrentVersionRepository currentVersionService,
+  ) = _UpdateRequiredService;
+
+  /// Checks if the application is running the minimum required version.
+  /// We use a Future<void> since this is a one-time check.
```

Comment:
This sentence is kind of weird in my opinion. Did you mean that every time we call this method, we do the whole app version validation and we should do it once per session?

Reply:
I changed the comment a bit